tests/stdlib/txmltree.nim

```diff
@@ -83,3 +83,19 @@ block:
   x.add newElement("sonTag")
   x.add newEntity("my entity")
   doAssert $x == "<myTag>my text<sonTag />&my entity;</myTag>"
+
+block: # bug #21290
+  let x = newXmlTree("foo",[
+    newXmlTree("bar",[
+      newText("Hola"),
+      newXmlTree("qux",[
+        newXmlTree("plugh",[])
+      ])
+    ])
+  ])
+
+  let s = $x
+  doAssert $parseXml(s) == s
+  doAssert s == """<foo>
+  <bar>Hola<qux>    <plugh />  </qux></bar>
```

Comment:
In the issue the expected output is `Hola<qux><plugh /></qux>`

Reply:
Yeah, I only agree that there shouldn't have any spaces after text nodes. That's an obvious bug since `$parseXml(s) == s` is not satisfied.


> Whatever the output is, there shouldn't be any spaces left between the text node and the consecutive node. So it should be:

Reply:
There are two proposed output in that issue

```xml
<foo>
  <bar>Hola  <qux>    <plugh />  </qux></bar>
</foo>
```

```xml
<foo>
  <bar>Hola<qux>
      <plugh />
    </qux>
  </bar>
</foo>
```